src/strands/tools/registry.py

```diff
@@ -538,6 +538,10 @@ def validate_tool_spec(self, tool_spec: ToolSpec) -> None:
                 }
                 continue
 
+            # It is expected that type and description are already included in referenced $def.
+            if "$ref" in prop_def:
```

Comment:
I *think* for structured output we actually flatten this out - see https://github.com/strands-agents/sdk-python/blob/684b3f7852776140a03a6dcf53e52a3cac4a5ef1/src/strands/tools/structured_output.py#L257

@theagenticguy as part of structured-output, do you happen to know why we flattened instead of using `$def`s and `$ref`s in the input schema?